operator/pkg/controlplane/apiserver/apiserver.go

```diff
@@ -0,0 +1,132 @@
+package apiserver
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+
+	operatorv1alpha1 "github.com/karmada-io/karmada/operator/pkg/apis/operator/v1alpha1"
+	"github.com/karmada-io/karmada/operator/pkg/constants"
+	"github.com/karmada-io/karmada/operator/pkg/controlplane"
+	"github.com/karmada-io/karmada/operator/pkg/util"
+)
+
+// EnsureKarmadaApiserver create karmada apiserver deployment and service resource
+func EnsureKarmadaApiserver(client clientset.Interface, cfg *operatorv1alpha1.KarmadaComponents, name, namespace string) error {
+	server, err := karamdaApiserver(client, cfg.KarmadaAPIServer, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	return server.Apply(client)
+}
+
+// EnsureKarmadaAggregratedAPIServer create karmada aggregated apiserver deployment and service resource
+func EnsureKarmadaAggregratedAPIServer(client clientset.Interface, cfg *operatorv1alpha1.KarmadaComponents, name, namespace string) error {
+	server, err := karmadaAggregratedAPIServer(client, cfg.KarmadaAggregratedAPIServer, name, namespace)
+	if err != nil {
+		return err
+	}
+	return server.Apply(client)
+}
+
+func karamdaApiserver(client clientset.Interface, cfg *operatorv1alpha1.KarmadaAPIServer, name, namespace string) (*controlplane.Server, error) {
```

Comment:
```suggestion
func karamdaAPIserver(client clientset.Interface, cfg *operatorv1alpha1.KarmadaAPIServer, name, namespace string) (*controlplane.Server, error) {
```
By the way, the `client` parameter is not used.